.github/workflows/spelling.yml

```diff
@@ -1,154 +1,52 @@
       with:
-        suppress_push_for_open_pull_request: 1
-        checkout: true
-        check_file_names: 1
-        spell_check_this: rancher-sandbox/rancher-desktop@main
-        post_comment: 0
-        use_magic_file: 1
-        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'rancher-sandbox' && 1 }}
-        report-timing: 1
-        warnings: bad-regex,binary-file,deprecated-feature,large-file,limited-references,no-newline-at-eof,noisy-file,non-alpha-in-dictionary,token-is-substring,unexpected-line-ending,whitespace-in-dictionary,minified-file,unsupported-configuration,no-files-to-check
-        use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
-        extra_dictionary_limit: 20
-        extra_dictionaries:
-          cspell:software-terms/dict/softwareTerms.txt
-          cspell:k8s/dict/k8s.txt
-          cspell:node/dict/node.txt
-          cspell:aws/aws.txt
-          cspell:golang/dict/go.txt
-          cspell:php/dict/php.txt
-          cspell:python/src/python/python-lib.txt
-          cspell:typescript/dict/typescript.txt
-          cspell:npm/dict/npm.txt
-          cspell:shell/dict/shell-all-words.txt
-          cspell:html/dict/html.txt
-          cspell:filetypes/filetypes.txt
-          cspell:fullstack/dict/fullstack.txt
-          cspell:python/src/common/extra.txt
-          cspell:java/src/java.txt
-          cspell:dotnet/dict/dotnet.txt
-          cspell:css/dict/css.txt
-          cspell:django/dict/django.txt
-          cspell:docker/src/docker-words.txt
-          cspell:cpp/src/stdlib-cmath.txt
-          cspell:python/src/python/python.txt
-          cspell:powershell/dict/powershell.txt
+        persist-credentials: false
 
-  update:
-    name: Update PR
-    permissions:
-      contents: write
-      pull-requests: write
-      actions: read
-    runs-on: ubuntu-latest
-    if: ${{
-        github.repository_owner != 'rancher-sandbox' &&
-        github.event_name == 'issue_comment' &&
-        github.event.issue.pull_request &&
-        contains(github.event.comment.body, '@check-spelling-bot apply')
-      }}
-    concurrency:
-      group: spelling-update-${{ github.event.issue.number }}
-      cancel-in-progress: false
-    steps:
-    - name: apply spelling updates
-      uses: check-spelling/check-spelling@2c9e4a85102fa9b6df3cb8bb5a8dc8bdc2fb2fea # v0.0.23
+    - uses: ./.github/actions/yarn-install
+    - uses: ./.github/actions/setup-environment
+    - run: sudo apt-get install cpanminus
+
+    - name: Check Spelling
+      run: yarn lint:spelling
+      env:
+        GITHUB_TOKEN: ${{ github.token }} # for SARIF uploading
+
+    - name: Upload SARIF report
+      # Use the limited report since if we have more than 25k errors nobody is
+      # going read through it all anyway.
+      if: always() && env.UPLOAD_SARIF_LIMITED != ''
```

Comment:
I don't understand where `UPLOAD_SARIF_LIMITED` is set; isn't it always unassigned?

Reply:
`yarn lint:spelling` (that is, `check-spelling`'s `unknown-words.sh`) looks at `$GITHUB_ENV` and outputs it to that file (which gets picked up in subsequent steps).

I'll add a comment.